common/djangoapps/student/models.py

```diff
@@ -823,10 +823,17 @@ def is_course_full(self, course):
 
     def users_enrolled_in(self, course_id):
         """Return a queryset of User for every user enrolled in the course."""
-        return User.objects.filter(
+        course_users = User.objects.filter(
```

Comment:
@dcadams, this is how I would have done it:
https://github.com/Stanford-Online/edx-platform/commit/a11740fbd783b8067bd472c66f7be82c71838f7b
Here I incorporate the implementation of UserProfile.has_registered into the filter that forms my initial query. However @stvstnfrd, informed me that the underlying implementation of direct access will be changing in the near future, and asked me to use the UserProfile.has_registered method, so that code would only have to change once, after direct access is reimplemented.